airbyte-integrations/bases/base-typing-deduping-test/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/BaseSqlGeneratorIntegrationTest.java

```diff
@@ -718,20 +718,58 @@ public void weirdColumnNames() throws Exception {
         dumpFinalTableRecords(streamId, ""));
   }
 
+  /**
+   * A stream with no columns is weird, but we shouldn't treat it specially in any way.
+   * It should create a final table as usual, and populate it with the relevant metadata columns.
+   */
+  @Test
+  public void noColumns() throws Exception {
+    createRawTable(streamId);
+    insertRawTableRecords(
+        streamId,
+        List.of(Jsons.deserialize(
+            """
+                {
+                  "_airbyte_raw_id": "14ba7c7f-e398-4e69-ac22-28d578400dbc",
+                  "_airbyte_extracted_at": "2023-01-01T00:00:00Z",
+                  "_airbyte_data": {}
```

Comment:
would _airbyte_data be null in this case? I have no idea

Reply:
:shrug:

my theory is that the data actually has random columns that aren't declared in the schema. And the protocol declares `data` as required, so I think we're ok to assume a nonnull object https://github.com/airbytehq/airbyte-protocol/blob/main/protocol-models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml#L55-L61

Reply:
also we're currently declaring `_airbyte_data` as `not null` on the raw tables >.>